src/java.base/share/classes/jdk/internal/util/ArraysSupport.java

```diff
@@ -160,6 +162,105 @@ public static int vectorizedMismatch(Object a, long aOffset,
         }
     }
 
+    // Possible values for the type operand of the NEWARRAY instruction.
+    // See https://docs.oracle.com/javase/specs/jvms/se9/html/jvms-6.html#jvms-6.5.newarray.
+
+    public static final int T_BOOLEAN = 4;
```

Comment:
As an idea for a follow-up enhancement, unless there are plans to implement runtime dispatching between different stubs, the basic type can be coded as a Class and on compiler side the corresponding basic type extracted with `java_lang_Class::as_BasicType()`.